### PR TITLE
prismlauncher: add some missing dependencies

### PR DESCRIPTION
**Pull request by Scrumplex**
###### Description of changes

Adding some missing dependencies, both for the launcher and the game itself. Also add a parameter to conveniently add more programs to PATH.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Scrumplex** [comment]
@ofborg eval

**Review by getchoo — CHANGES_REQUESTED** [review]
for narrator mode, flite is detected but no audio is actually played. not really sure what's going on here 🤔

**Comment by getchoo** [comment]
found the issue! i've packaged prism with flite on fedora and it worked there, and the only difference i noticed was that flite currently builds against alsa in nixpkgs, while fedora builds against pulse. using the following repl commands and running prism again, flite was able to work :)

image

i've opened https://github.com/NixOS/nixpkgs/pull/239194 to create a new `flite-pulse` package, but if that doesn't get merged we could just use the overrides i made above

**Comment by Scrumplex** [comment]
Let's wait for #239194 to be reviewed and merged

**Comment by getchoo** [comment]
https://github.com/NixOS/nixpkgs/pull/239194 has been merged, but we should probably wait for it to get through staging :p

**Comment by Scrumplex** [comment]
Now that https://github.com/NixOS/nixpkgs/pull/239194 has reached master, I rebased this PR.

Narrator now works!

Not sure if there is anything we need to add for macOS though

**Comment by getchoo** [comment]
> Not sure if there is anything we need to add for macOS though

i'm not sure what the macos version uses, but since it worked with the current expression i'm guessing it's something pre installed with the system - so nothing should be needed

**Review by getchoo — APPROVED** [review]
worked for me as well!

**Comment by getchoo** [comment]
Result of `nixpkgs-review pr 238645` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>prismlauncher</li>
    <li>prismlauncher-qt5</li>
  </ul>
</details>

File: pkgs/games/prismlauncher/wrapper.nix [hunk 1]
```diff
@@ -61,14 +65,21 @@ symlinkJoin {
         stdenv.cc.cc.lib
       ]
       ++ lib.optional gamemodeSupport gamemode.lib
+      ++ lib.optional textToSpeechSupport flite
       ++ additionalLibs;
 
+      programs = [
```
Comment (SuperSandro2000) [inline pkgs/games/prismlauncher/wrapper.nix:71]: A more widely used name would be runtimeDeps

**Pull request merged**